meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common-alternative.cmd.in

```diff
@@ -145,7 +187,7 @@ setenv bootcmd_rollback 'if test -n "${kernel_image2}" && test "${fiovb.rollback
 if test "${fiovb.is_secondary_boot}" = "1"; then
 	# and it became corrupted after update, which means
 	# that we should rollback also
-	if test "${fiovb.bootupgrade_available}" = "1"; then
+	if test "${fiovb.bootupgrade_primary_updated}" = "1"; then
```

Comment:
IMHO, we should check `upgrade_available` to determine if we `run rollback_setup` (both being rootfs focused).
I'll leave it up to you, tho.  I don't feel that strongly either way.